### PR TITLE
[THOG-709] - Recover from detector panics

### PR DESCRIPTION
**Pull request by ahrav**

### CODE REVIEW

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -163,7 +163,12 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 	for chunk := range e.chunks {
 		fragStart, mdLine := fragmentFirstLine(chunk)
 		for _, decoder := range e.decoders {
-			decoded := decoder.FromChunk(chunk)
+			decoded := func(ctx context.Context) *sources.Chunk {
+				defer func(ctx context.Context) {
+					common.Recover(ctx)
```
Comment (dustin-decker) [inline pkg/engine/engine.go:168]: This function calls os.Exit(1), maybe we should update it to optionally exit

Comment (mcastorina) [reply pkg/engine/engine.go:168]: Maybe renaming `Recover` to `RecoverAndExit` or something would make it more clear, then have `Recover` not exit.

Comment (dustin-decker) [reply pkg/engine/engine.go:168]: Yes, let's do that

Comment (dustin-decker) [reply pkg/engine/engine.go:168]: Also the integration tests are fixed in main, please update your branch

**Review by mcastorina — APPROVED** [review]

**Review by dustin-decker — APPROVED** [review]

**Pull request merged**